interfaces/devdoc/threadpool_requirements.md

```diff
@@ -60,35 +62,45 @@ MOCKABLE_FUNCTION(, THREADPOOL_HANDLE, threadpool_create, EXECUTION_ENGINE_HANDL
 
 **NON_THREADPOOL_01_002: [** If `execution_engine` is `NULL`, `threadpool_create` shall fail and return `NULL`. **]**
 
+[Will be moved from threadpool_open.]
+**N_O_N_THREADPOOL_01_011: [** `threadpool_open` shall switch the state to OPENING and perform any actions needed to open the `threadpool` object. **]**
+
 **NON_THREADPOOL_01_003: [** If any error occurs, `threadpool_create` shall fail and return `NULL`. **]**
 
 ### threadpool_destroy
 
 ```c
-MOCKABLE_FUNCTION(, void, threadpool_destroy, THREADPOOL_HANDLE, threadpool);
+MOCKABLE_FUNCTION(, void, threadpool_dispose, THREADPOOL_HANDLE, threadpool);
 ```
 
-`threadpool_destroy` frees all resources associated with `threadpool`.
+`threadpool_dispose` frees all resources associated with `threadpool`.
+
+**NON_THREADPOOL_01_004: [** If `threadpool` is `NULL`, `threadpool_dispose` shall return. **]**
 
-**NON_THREADPOOL_01_004: [** If `threadpool` is `NULL`, `threadpool_destroy` shall return. **]**
+**NON_THREADPOOL_01_005: [** Otherwise, `threadpool_dispose` shall free all resources associated with `threadpool`. **]**
 
-**NON_THREADPOOL_01_005: [** Otherwise, `threadpool_destroy` shall free all resources associated with `threadpool`. **]**
+**NON_THREADPOOL_01_006: [** While `threadpool` is OPENING, `threadpool_dispose` shall wait for the open to complete either successfully or with error. **]**
 
```

Comment:
"How do you kill the one which has no life?" :-)
How can it happen to call dispose on something that was not created yet?

Reply:
I think it waits for OPENING to become OPEN if there is a rapid create and destroy transitioning.

Reply:
Fixed.